### ISSUE TITLE
Trap gamemode change events

### ISSUE THREAD

**Issue by adarshp** [ml4ai/tomcat#157]
We should implement a handler to trap `CommandEvent`s corresponding to players trying to change the gamemode from adventure mode to survival or creative mode, and also a an option to disable the trap for developers.

**Comment by wilsonliam** [comment]
Completed and new branch pushed; not sure if I should introduce a new 'whitelist' json in this before PR.

**Comment by adarshp** [comment]
Yes, we'll need a whitelist for sure. Let's discuss in more detail at the meeting.

Also, the current implementation results in the JSON config file being parsed every time there is a CommandEvent. File I/O is relatively slow and should be minimized as much as possible, so it would be better to parse it just once, at the time of creation of the `ForgeEventHandler` singleton instance, store the result in a boolean member variable, and access it within the `handle` method that handles `CommandEvent` events.

**Comment by wilsonliam** [comment]
Just pushed a new branch with a better implementation and whitelist included (with the only two defaults I believe are needed)...file IO happens once and it's more readable.
